mypy/subtypes.py

```diff
@@ -204,7 +203,8 @@ def visit_tuple_type(self, left: TupleType) -> bool:
                 if right.args:
                     iter_type = right.args[0]
                 else:
-                    iter_type = AnyType()
+                    # todo: is type of this Any correct?
+                    iter_type = AnyType(TypeOfAny.from_error)
```

Comment:
@ilevkivskyi, hmm I see. I think it's better to have a separate PR for that one. This doesn't seem to be relevant to this PR and this PR is already huge.